packages/agreement-process/src/services/readModelService.ts

```diff
@@ -587,6 +610,36 @@ export function readModelServiceBuilder(
       }
       return result.data;
     },
+
+    async getActiveConsumerDelegationByEserviceAndDelegatorIds(
+      eserviceId: EServiceId,
+      delegatorId: TenantId
+    ): Promise<Delegation | undefined> {
+      return getDelegation(delegations, {
+        "data.eserviceId": eserviceId,
+        "data.delegatorId": delegatorId,
+        "data.state": delegationState.active,
+        "data.kind": delegationKind.delegatedConsumer,
+      });
+    },
+
+    async getActiveConsumerDelegationByEserviceAndIds({
+      eserviceId,
+      delegatorId,
+      delegateId,
+    }: {
+      eserviceId: EServiceId;
+      delegatorId: TenantId;
+      delegateId: TenantId;
+    }): Promise<Delegation | undefined> {
+      return getDelegation(delegations, {
+        "data.eserviceId": eserviceId,
+        "data.delegatorId": delegatorId,
+        "data.delegateId": delegateId,
+        "data.state": delegationState.active,
+        "data.kind": delegationKind.delegatedConsumer,
+      });
+    },
```

Comment:
> What about joining these two methods with name `getActiveConsumerDelegation`?

I got rid of them alltogether, since were used only once, let me know what you think!